src/test/java/ui/BaseUITest.java

```diff
@@ -0,0 +1,66 @@
+package ui;
+
+import api.ApiClient;
+import controllers.WidgetController;
+import exeptions.FailedToLoginException;
+import helpers.UserLogin;
+import org.apache.hc.core5.http.HttpStatus;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import ui.screenshotService.ScreenshotWatcher;
+import ui.api.ApiSteps;
+import ui.driver.DriverManager;
+import ui.driver.wait.CustomFluentWait;
+import ui.pages.LoginPage;
+
+import static ui.logger.UILogger.LOGGER;
+
+public class BaseUITest {
+    private static ApiClient api = new ApiClient();
+    private static UserLogin userLogin = new UserLogin(api);
+    protected static ApiSteps apiSteps = new ApiSteps();
+    protected static WidgetController widget = new WidgetController(api);
+    protected static int widgetId1;
+    protected static int widgetId2;
+
+
+    @BeforeAll
+    public static void setupDriver() throws FailedToLoginException {
+        userLogin.login();
+        DriverManager.setupDriver();
+        CustomFluentWait.setCustomFluentWait();
+        LOGGER.info("Opening login page and login ...");
+        var loginPage = new LoginPage();
```

Comment:
Is it a necessary variable? Maybe better to use new LoginPage().openLoginPage().enterCredentials().clickLoginButton();

Reply:
removed variable